src/psyclone/tests/psyir/transformations/inline_trans_test.py

```diff
@@ -1646,10 +1646,50 @@ def test_validate_unknowntype_var(fortran_reader):
     inline_trans = InlineTrans()
     with pytest.raises(TransformationError) as err:
         inline_trans.validate(routine)
-    assert ("Routine 'sub' cannot be inlined because it contains a Symbol "
-            "'x' which is of unknown type: 'REAL, POINTER, INTENT(INOUT) :: x'"
+    assert ("Routine 'sub' cannot be inlined because it contains a Symbol 'x' "
+            "which is an Argument of UnknownType: 'REAL, POINTER, "
+            "INTENT(INOUT) :: x'" in str(err.value))
+
+
+def test_validate_unknowninterface(fortran_reader, fortran_writer):
+    '''
+    Test that validate rejects a subroutine with arguments of UnknownType.
+
+    '''
+    code = (
+        "module test_mod\n"
+        "contains\n"
+        "subroutine main\n"
+        "  call sub()\n"
+        "end subroutine main\n"
+        "subroutine sub()\n"
+        "  real, pointer :: x\n"
+        "  x = x + 1.0\n"
+        "end subroutine sub\n"
+        "end module test_mod\n"
+    )
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    with pytest.raises(TransformationError) as err:
+        inline_trans.validate(routine)
+    assert (" Routine 'sub' cannot be inlined because it contains a Symbol "
+            "'x' with an UnknownInterface: 'REAL, POINTER :: x'"
             in str(err.value))
 
+    # But if the interface is known, it has no problem inlining it
+    xvar = psyir.walk(Routine)[1].symbol_table.lookup("x")
+    xvar.interface = AutomaticInterface()
+    inline_trans.apply(routine)
+    assert fortran_writer(psyir.walk(Routine)[0]) == """\
+subroutine main()
+  REAL, POINTER :: x
+
+  x = x + 1.0
+
+end subroutine main
+"""
+
```

Comment:
Can we check that the output compiles?

Also, and possibly not your fault, pylint says that `tmpdir` is unused at L608.

Reply:
Added compilation for this test and removed L608 tmpdir